bam_mate.c

```diff
@@ -155,9 +155,27 @@ static bool plausibly_properly_paired(bam1_t* a, bam1_t* b)
         return false;
 }
 
-static void sync_mq(bam1_t* src, bam1_t* dest)
+static void bam_format_cigar(const bam1_t* b, kstring_t* str)
+{
+    // An empty cigar is a special case return "*" rather than ""
+    if (b->core.n_cigar == 0) {
+        kputc('*', str);
+        return;
+    }
+
+    const uint32_t *cigar = bam_get_cigar(b);
+    uint16_t i;
+
+    for (i = 0; i < b->core.n_cigar; ++i) {
+        kputw(bam_cigar_oplen(cigar[i]), str);
```

Comment:
Kstring since https://github.com/samtools/htslib/commit/67234cb0115001eb92909726fc7259bf1806377a now returns error codes, which means  bam_format_cigar() should check the return value from kputw and kputc.  If feels a bit churlish to point this out given this also means sync_mq_mc should return a value, which in turn means sync_mate should return a value too, and they aren't entirely new functions added by this PR.  However as other changes  would be useful I'll mention this here too. :-)